### DISCUSSION TITLE
Root table of references in GC

### DISCUSSION THREAD

**Discussion by ShervanN** [ProfessionalCSharp/ProfessionalCSharp2021#133] [General]
Hi,

Page 340, CHAPTER 13   Managed and Unmanaged Memory, below Garbage Collection you wrote "The GC finds all referenced objects from a **root table of references** and continues to a tree of referenced objects." 
But didn't explain about  **root table of references**

**Comment by christiannagel** [comment]
When you create an object, e.g. from this `Main` method, the references `p1` and `p2` are added to the root table of references. 

```csharp
Person p1 = new();
Person p2 = new();
//...
```

If the `Person` object itself creates an inner object, such as an `Address` object, which is stored in a member of the `Person` type, the reference to the address is not stored on the root table. This reference is found in the tree of referenced objects.
